cardano-cli/test/cardano-cli-golden/Test/Golden/CreateTestnetData.hs

```diff
@@ -191,3 +191,37 @@ hprop_golden_create_testnet_data_deleg_non_deleg =
     -- https://github.com/IntersectMBO/cardano-cli/issues/631 witnesses.
 
     H.assertWith (fromIntegral onlyHolderCoin) (\ohc -> ohc > totalSupply `div` 2)
+
+-- Execute this test with:
+-- @cabal test cardano-cli-golden --test-options '-p "/golden create testnet data total supply template/"'@
+--
+-- This test checks that we use the total supply value from the shelley template when --total-supply
+-- is not specified. It was broken in the past (see https://github.com/IntersectMBO/cardano-node/issues/5953).
+hprop_golden_create_testnet_data_total_supply_template :: Property
+hprop_golden_create_testnet_data_total_supply_template =
+  propertyOnce $ moduleWorkspace "tmp" $ \tempDir -> do
+    vanillaShelleyGenesis :: ShelleyGenesis StandardCrypto <-
+      H.readJsonFileOk "test/cardano-cli-golden/files/input/shelley/genesis/genesis.spec.json"
+    let tweakedValue = 3_123_456_000_000
+        tweakedShelleyGenesis = vanillaShelleyGenesis{L.sgMaxLovelaceSupply = tweakedValue}
+        tweakedShelleyGenesisFp = tempDir </> "tweaked-shelley-genesis.json"
+
+    void $ liftIO $ writeFileJSON tweakedShelleyGenesisFp tweakedShelleyGenesis
+
+    let outputDir = tempDir </> "out"
+    void $
+      execCardanoCLI
+        [ "conway"
+        , "genesis"
+        , "create-testnet-data"
+        , "--spec-shelley"
+        , tweakedShelleyGenesisFp
+        , "--out-dir"
+        , outputDir
+        ]
+
+    outputGenesis :: ShelleyGenesis StandardCrypto <-
+      H.readJsonFileOk $ outputDir </> "shelley-genesis.json"
+
+    -- Because we don't test this elsewhere in this file:
+    (L.sgMaxLovelaceSupply outputGenesis) H.=== tweakedValue
```

Comment:
Rather than checking a single value, why not do a golden test on the entire genesis value?

Reply:
@Jimbo4350> I hadn't done it initially, because the generated genesis file is not stable.

However, I brought [cardano-node's redactJsonFieldsInFile](https://github.com/IntersectMBO/cardano-node/blob/master/cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Query.hs#L428) function to redact the unstable fields, and I now made this a golden test.

Reply:
@Jimbo4350> I did that in an additional commit that I will fixup, to make your review easier.